src/components/catalog-search/catalog-search.controller.ts

```diff
@@ -42,6 +42,12 @@ export class CatalogSearchController implements angular.IController {
     }
   }
 
+  public onKeyPress = (keyEvent: any) => {
+    if (keyEvent.which === 13) {
+      this.$rootScope.$emit('filter-catalog-items', {searchText: this.ctrl.searchText});
```

Comment:
What happens if there is no search text?

Reply:
Nothing :-)  No filter is applied in the service-view/browse catalog

Reply:
If I had a previous search, is it cleared?

I think I'd rather check 

```js
if (keyEvent.which === 13 && this.ctrl.searchText)
```

here to avoid event firing the event. Then we're not relying on any specific service view search behavior.

Reply:
> If I had a previous search, is it cleared?

No, it remains
I'll add the other condition, thanks

Reply:
ok, updated

Reply:
great, thanks @dtaylor113 !